From git-commits-head-owner@vger.kernel.org Thu Nov 30 20:59:09 2006
Date: Wed, 29 Nov 2006 21:59:04 GMT
Message-Id: <200611292159.kATLx4Hw020601@hera.kernel.org>
From: Linux Kernel Mailing List <linux-kernel@vger.kernel.org>
To: git-commits-head@vger.kernel.org
Subject: [PATCH] r8169: Fix iteration variable sign
Content-Length: 1397
Lines: 37

Gitweb:     http://git.kernel.org/git/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commit;h=315917d23fdd20a0f4ff99b9228de5840d9d276c
Commit:     315917d23fdd20a0f4ff99b9228de5840d9d276c
Parent:     af768c6711de615abf9c679732a2f8245a4c286c
Author:     Francois Romieu <romieu@fr.zoreil.com>
AuthorDate: Wed Nov 29 22:21:33 2006 +0100
Committer:  Linus Torvalds <torvalds@woody.osdl.org>
CommitDate: Wed Nov 29 13:45:07 2006 -0800

    [PATCH] r8169: Fix iteration variable sign
    
    This changes the type of variable "i" in rtl8169_init_one()
    from "unsigned int" to "int". "i" is checked for < 0 later,
    which can never happen for "unsigned". This results in broken
    error handling.
    
    Signed-off-by: Michael Buesch <mb@bu3sch.de>
    Signed-off-by: Francois Romieu <romieu@fr.zoreil.com>
    Signed-off-by: Linus Torvalds <torvalds@osdl.org>
---
 drivers/net/r8169.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/r8169.c b/drivers/net/r8169.c
index 27f90b2..b977ed8 100644
--- a/drivers/net/r8169.c
+++ b/drivers/net/r8169.c
@@ -1473,8 +1473,8 @@ rtl8169_init_one(struct pci_dev *pdev, c
 	struct rtl8169_private *tp;
 	struct net_device *dev;
 	void __iomem *ioaddr;
-	unsigned int i, pm_cap;
-	int rc;
+	unsigned int pm_cap;
+	int i, rc;
 
 	if (netif_msg_drv(&debug)) {
 		printk(KERN_INFO "%s Gigabit Ethernet driver %s loaded\n",

